lib/net/ssh/transport/session.rb

```diff
@@ -61,12 +61,12 @@ def initialize(host, options={})
       @bind_address = options[:bind_address] || nil
       @options = options
 
-      debug { "establishing connection to #{@host}:#{@port}" }
-
       @socket =
         if (factory = options[:proxy])
+          debug { "establishing connection to proxy" }
```

Comment:
Ok, deeper dive on the happy path:

1. Make TCP connection to proxy (start of the factory call)
1. SSH auth to proxy (what successive debug messages from the factory may talk about)
1. Open channel to proxy over established connection 
1. Send direct-tcpip request (https://tools.ietf.org/html/rfc4254#section-7.2)
  a. proxy attempts to connect to some endpoint
  b. proxy replies with, "ok, this channel now represents a connection to remote port"
  c. (factory returns channel as if it's a socket)
1. SSH auth + requests to the destination host

When attempting to debug interactions with a proxy, it's actually step 4 that is making a connection to the destination host. Given that steps 1,2,3 can all fail before step 4 is attempted, simply stating "connecting to #{@host}:#{@port} through proxy" here is somewhat misleading.

In the best scenario, we might consider passing debug level flags into the factory so that we can illuminate steps 1-4 in which case this debug message goes away entirely.

Like I said earlier, this message (and really any variation that anyone have mentioned) are a vast improvement over current behavior.

If we want to pull from OpenSSH logging behavior as a model, then:

```
$ ssh -v -o ProxyCommand="ssh -W %h:1 myproxy.example.com" destination.example.com
OpenSSH_7.2p2, LibreSSL 2.4.1
debug1: Reading configuration data /Users/imoverclocked/.ssh/config
debug1: /Users/imoverclocked/.ssh/config line 38: Applying options for *
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 20: Applying options for *
debug1: Executing proxy command: exec ssh -W destination.example.com:1 myproxy.example.com
...
```

```
$ ssh -v -W destination.example.com:1 myproxy.example.com
OpenSSH_7.2p2, LibreSSL 2.4.1
debug1: Reading configuration data /Users/imoverclocked/.ssh/config
debug1: /Users/imoverclocked/.ssh/config line 38: Applying options for *
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 20: Applying options for *
debug1: Connecting to myproxy.example.com [1.1.1.1] port 22.
debug1: fd 3 clearing O_NONBLOCK
debug1: Connection established.
... auth stuff ...
debug1: channel_connect_stdio_fwd destination.example.com:1
debug1: channel 0: new [stdio-forward]
debug1: getpeername failed: Bad file descriptor
... connection shutdown ...
```

ie: the top level ssh process mentions that it is starting a proxy command and it's up to the proxy (or proxy command in this case) to talk about connecting to destination.example.com:1